Player data types updated and saved to server on restart

Jayefuu opened the pull request:
- Saves more kill data to Scoreboard
- Displays total kills in score board
- Saves player data to crash site scenario data set
- Added scenario name to data set so that if we later want to add data for danger ores we can filter it out

Jayefuu committed:
Player data types updated

- Saves more kill data to Scoreboard
- Displays total kills in score board
- Saves player data to crash site scenario data set
- Added scenario name to data set so that if we later want to add data for danger ores we can filter it out

grilledham committed:
grilledham's review changes:

- Rename biter_kill_counter->entity_died
- Refactor logic in entity_died function to reduce table lookups.
- Fix nil cause bug.
- Fix player_list kills header different size to cells.

grilledham commented on features/player_stats.lua:
```diff
@@ -137,9 +174,30 @@ local function robot_built_entity()
+    change_for_global(aliens_killed_name, 1)
+
+    local cause = event.cause
+    if not cause or not cause.valid or cause.name ~= 'character' then -- What about players inside vehicles?
```
I left this comment
> What about players inside vehicles?

Do we want to record this too, if so I've left it as an exercise for the reader.

Jayefuu replied:
Fixed! I haven't managed deaths caused by manually driving trains but I don't think that's necessary.

grilledham replied:
Nice!

I think that is much better, we can always add the maunal trains kills later if we want.

grilledham commented on features/player_stats.lua:
```diff
@@ -21,6 +21,11 @@ local player_console_chats_name = 'player-console-chats'
+local player_worms_killed_name = 'player-worms-killed'
+local player_spawners_killed_name = 'player-spawners-killed'
+local player_total_kills_name = 'player-total-kills'
+local player_turrets_killed_name = 'player-turrets-killed'
```
I wonder if we should expose these variables publicly, so they can be used in command.lua.
But don't worry if it's too much effort.

Jayefuu replied:
How do I do this and what's the advantage?

grilledham replied:
I made the change, so you can see it in the commit.

The advantage is that if the same values are declared separately they can go out of sync, which if that ever happened would be a bug. So it's better to have a single source of truth for values that are logically the same thing.

grilledham commented:
I made some changes, but I must confess I was lazy and did minimal testing.

grilledham commented on features/gui/player_list.lua:
```diff
@@ -498,16 +531,17 @@ local function get_default_player_settings()
+
     return {
         columns = columns,
         sort = -3
```
You changed this value to -4, so I changed it back to -3, did you mean to change it?

Jayefuu replied:
Woops.

grilledham committed:
More player_list tweaks.

Jayefuu commented on features/player_stats.lua:
```diff
@@ -137,9 +174,30 @@ local function robot_built_entity()
+    local player_index = cause.player.index
+    change_for_player(player_index, player_total_kills_name, 1)
+
+    if category then
```
Why do this? We already return if category is nil.....

grilledham replied:
category could be `false`. At least the way I setup the table.

grilledham commented on map_gen/maps/crash_site/commands.lua:
```diff
@@ -90,22 +90,42 @@ function Public.control(config)
+            end
+
             local statistics = {
+                scenario = data.scenario_name,
```
```suggestion
                scenario = config.scenario_name,
```

`data.scenario_name` is the scenario that is going to be started. Could be different if an admin used the restart command.

Jayefuu committed:
Updated as per suggestions

- Added small fixes to commands.lua
- Added logic to player_stats.lua to count kills made with vehicles. Tested on S10 with a passenger.

grilledham committed:
Refactor get player index logic

grilledham committed:
Expose PlayerStats names publicly.

grilledham commented:
I made some changes:

- Refactored the get player index from cause logic. Moved into a function and reordered to reduce the amount of nested if blocks.
- Fixed the merge conflict.
- Made the player stat names public.

grilledham committed:
Use PlayerStat names in player_list.lua
